eng/common/pipelines/templates/steps/verify-agent-os.yml

```diff
@@ -0,0 +1,38 @@
+# Template for all Python Scripts in this repository
+parameters: 
+  OSVmImage: $(OSVmImage)
+
+steps:
+  - task: PythonScript@0
+    displayName: Verify Agent OS
+    inputs:
+      scriptSource: inline
+      script: |
+        # Script verifies the operating system for the platform on which it is being run
+        # Used in build pipelines to verify the build agent os
+        # Variable: The friendly name or image name of the os to verfy against
+        from __future__ import print_function
+        import sys
+        import platform
+
+        os_parameter = "${{ parameters.OSVmImage }}".lower()
+        if os_parameter.startswith('mac') or os_parameter.startswith('darwin'):
+            os_parameter = 'macOS'
+        elif os_parameter.startswith('ubuntu') or os_parameter.startswith('linux'):
+            os_parameter = 'Linux'
+        elif os_parameter.startswith('vs') or os_parameter.startswith('win'):
+            os_parameter = 'Windows'
+        else:
+            raise Exception('Variable OSVmImage is empty or has an unexpected value')
```

Comment:
For better debugging you should output the value.

`Variable OSVmImage is empty or has an unexpected value [${{ parameters.OSVmImage }}]`